extras/bindings/c/include/precice/SolverInterfaceC.h

```diff
@@ -460,6 +460,35 @@ const char *precicec_actionReadIterationCheckpoint();
  */
 ///@{
 
+/// @copydoc precice::SolverInterface::isGradientDataRequired())
+int precicec_isGradientDataRequired(int dataID);
+
+/// @copydoc precice::SolverInterface::writeScalarGradientData()
+void precicec_writeScalarGradientData(
+    int           dataID,
+    int           valueIndex,
+    const double *gradientValues);
+
+/// @copydoc precice::SolverInterface::writeBlockScalarGradientData
+void precicec_writeBlockScalarGradientData(
+    int           dataID,
+    int           size,
+    const int *   valueIndices,
+    const double *gradientValues);
+
+/// @copydoc precice::SolverInterface::writeVectorGradientData()
+void precicec_writeVectorGradientData(
+    int           dataID,
+    int           valueIndex,
+    const double *gradientValues);
+
+/// @copydoc precice::SolverInterface::writeBlockVectorGradientData())
```

Comment:
Not sure what the best practice is for [copydoc](https://www.doxygen.nl/manual/commands.html#cmdcopydoc), but in the `isGradientDataRequired())` and `writeBlockVectorGradientData())` there is an extra closing parenthesis, while in the `writeBlockScalarGradientData` there is none. In [SolverInterfaceImpl.hpp](https://github.com/precice/precice/blob/develop/src/precice/impl/SolverInterfaceImpl.hpp), we use parentheses with types for any overloaded functions, and we don't add any parentheses otherwise. Let's keep that style.

```suggestion
/// @copydoc precice::SolverInterface::isGradientDataRequired
int precicec_isGradientDataRequired(int dataID);

/// @copydoc precice::SolverInterface::writeScalarGradientData
void precicec_writeScalarGradientData(
    int           dataID,
    int           valueIndex,
    const double *gradientValues);

/// @copydoc precice::SolverInterface::writeBlockScalarGradientData
void precicec_writeBlockScalarGradientData(
    int           dataID,
    int           size,
    const int *   valueIndices,
    const double *gradientValues);

/// @copydoc precice::SolverInterface::writeVectorGradientData
void precicec_writeVectorGradientData(
    int           dataID,
    int           valueIndex,
    const double *gradientValues);

/// @copydoc precice::SolverInterface::writeBlockVectorGradientData
```

Same in `precicec_isMeshConnectivityRequired`, as well as the complete `SolverInterfaceFortran.hpp`.